src/IndexItem.tsx

```diff
@@ -5,12 +5,14 @@ import type { IndexItemProps } from './types';
 
 class IndexItem extends PureComponent<IndexItemProps> {
   render() {
-    const { index, item, onSelectIndex, indexItemHeight, style } = this.props;
+    const { index, item, onSelectIndex, indexItemHeight, style, indexTextStyle } = this.props;
     return (
       <TouchableOpacity
         onPressIn={() => onSelectIndex({ index, item, method: 'press' })}
         style={[styles.indexItem, style, { height: indexItemHeight }]}>
-        <Text style={styles.indexText}>{item}</Text>
+        <Text style={indexTextStyle ? [styles.indexText, indexTextStyle] : styles.indexText}>
```

Comment:
there is no need for a condition here since React is ignoring falsey values in a style array :)

Reply:
updated.